core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -231,4 +310,91 @@ public void stopSync(AlluxioURI uri) { }
   protected static String validatePath(String path) {
     return new AlluxioURI(path).toString();
   }
+
+  /**
+   * Represents a under filesystem operation.
+   */
+  private interface UfsOperation<T> {
+    /**
+     * Applies this operation.
+     *
+     * @return the result of this operation
+     */
+    T apply() throws IOException;
+  }
+
+  /**
+   * Represents a under filesystem operation with a boolean return type.
+   */
+  private interface UfsBooleanOperation<Boolean> {
```

Comment:
no need for the `Boolean` generic argument - it doesn't get used anywhere

Reply:
Removed the boolean, `retryOnException` and `retryOnFalse` both use the `ObjectStoreOperation<T>`.